### PR TITLE
fix vfunc inconsistency between env_hdfs and env

### PR DESCRIPTION
**Pull request by shuzhang1989**
@siying 
This fixes the build failure of hdfs environment.

### CODE REVIEW

File: hdfs/env_hdfs.h [hunk 1]
```diff
@@ -101,7 +101,7 @@ class HdfsEnv : public Env {
                            std::shared_ptr<Logger>* result);
 
   virtual void Schedule(void (*function)(void* arg), void* arg,
-                        Priority pri = LOW, void* tag = nullptr) {
+                        Priority pri = LOW, void* tag = nullptr, void (*unschedFunction)(void* arg) = 0) {
     posixEnv->Schedule(function, arg, pri, tag);
```
Comment (IslamAbdelRahman) [inline hdfs/env_hdfs.h]: Can you update 
```posixEnv->Schedule(function, arg, pri, tag);```
to be
```posixEnv->Schedule(function, arg, pri, tag, unschedFunction);```

**Comment by IslamAbdelRahman** [comment]
Thanks @shuzhang1989

**Comment by shuzhang1989** [comment]
@IslamAbdelRahman updated.

**Comment by IslamAbdelRahman** [comment]
Thanks

**Pull request merged**